src/ast/passes/semantic_analyser.cpp

```diff
@@ -1682,6 +1683,35 @@ void SemanticAnalyser::binop_int(Binop &binop)
   }
 }
 
+void SemanticAnalyser::binop_array(Binop &binop)
+{
+  auto &lht = binop.left->type;
+  auto &rht = binop.right->type;
+  if (binop.op != Operator::EQ && binop.op != Operator::NE)
+  {
+    LOG(ERROR, binop.loc, err_)
+        << "The " << opstr(binop) << " operator cannot be used on arrays.";
+  }
+
+  if (!lht.GetElementTy()->IsIntegerTy() || !rht.GetElementTy()->IsIntegerTy())
```

Comment:
maybe we can just do:

```
  if (lht.GetNumElements() != rht.GetNumElements())
  {
    LOG(ERROR, binop.loc, err_)
     << "Arrays must be of the same length. " << lth << " != " << rht;
  }

if (!lht.GetElementTy()->IsIntegerTy()  || lht != rht) { // uses the IsEqual method on sized type
  LOG(ERROR, binop.loc, err_)
     << "Array elements must be of the same integer type. " << lth << " != " << rht;
}
```

This removes 2 related checks and also indicates what the mismatching types are

Reply:
Sure!